### ISSUE TITLE
Calendar, TreeSelect, MultiSelect, AutoComplete, CascadeSelect, Table | aria-expanded attribute is undefined and aria-controls attribute referencing a hidden element

### ISSUE THREAD

**Issue by TaneliTuomola** [primefaces/primeng#14358]
### Describe the bug

Required ARIA attribute is missing.

Elements with role "combobox" should contain aria-expanded and aria-controls attribute. Aria-expanded attribute is missing and aria-controls attribute refers to the element that is not in the DOM yet. Aria-controls only needs to be set when the popup is visible.

The issue affects at least a few form components that use overlay: Calendar, Dropdown & TreeSelect.

The same problem also applies to the Paginator Dropdown: [https://github.com/primefaces/primeng/issues/14260](https://github.com/primefaces/primeng/issues/14260)

Also related to the OverlayPanel aria issue: [https://github.com/primefaces/primeng/issues/14290](https://github.com/primefaces/primeng/issues/14290)

### Environment

Windows, Chrome

### Reproducer

https://primeng.org/dropdown

### Angular version

17.0.0

### PrimeNG version

17.0.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.0

**Comment by barbGit2** [comment]
In addition to what was mentioned in this ticket, the Calendar's **panelId** does not refer to any element in the Calender.  I think the Calendar template needs to be updated so that the **div.p-datepicker** element should have an **id** attribute set to **panelId**:
```
...
            <div
                #contentWrapper
           [attr.id]="panelId"
                [class]="panelStyleClass"
                [ngStyle]="panelStyle"
                [ngClass]="{
                    'p-datepicker p-component': true,
                    'p-datepicker-inline': inline,
                    'p-disabled': disabled,
                    'p-datepicker-timeonly': timeOnly,
                    'p-datepicker-multiple-month': this.numberOfMonths > 1,
                    'p-datepicker-monthpicker': view === 'month',
                    'p-datepicker-touch-ui': touchUI
                }"
...
```

**Comment by cetincakiroglu** [comment]
Hi,

Thanks for reporting the issue, we'll address all the overlay components with 17.9.0 release.
